src/basic/cgroup-util.c

```diff
@@ -860,6 +860,10 @@ int cg_pidref_get_path(const char *controller, const PidRef *pidref, char **ret_
         if (!pidref_is_set(pidref))
                 return -ESRCH;
 
+        // XXX: Ideally we'd use pidfd_get_cgroupid() + cg_path_from_cgroupid() here, to extract this
+        // bit of information from pidfd directly. However, the latter requires privilege and it's
+        // not entirely clear how to handle cgroups from outer namespace.
```

Comment:
we probably could try cg_path_from_cgroupid() and then fall back if it fails with EPERM.

i think the path for cgroups from outside of the namespace are marked specially, start with `../` or so something like that, so that one case recognize them. don't remember the details, but this was a solvable problem iirc

Reply:
I tried, and the path attached to fd is simply `/`...